### PR TITLE
Adopt ManagedCluster & Klusterlet API in nucleus

### PR DESCRIPTION
**Pull request by mdelder**

### CODE REVIEW

File: deploy/klusterlet/olm-catalog/klusterlet/manifests/0000_00_operator.open-cluster-management.io_klusterlets.crd.yaml~eeb95e939f074ff44e80dff94ef9973d34d8d28a [hunk 1]
```diff
@@ -0,0 +1,126 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (qiujian16) [inline deploy/klusterlet/olm-catalog/klusterlet/manifests/0000_00_operator.open-cluster-management.io_klusterlets.crd.yaml~eeb95e939f074ff44e80dff94ef9973d34d8d28a]: i think this file should be removed?

File: deploy/klusterlet/olm-catalog/klusterlet/manifests/0000_00_operator.open-cluster-management.io_klusterlets.crd.yaml~Adopt new API updates for operator [hunk 2]
```diff
@@ -0,0 +1,126 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (qiujian16) [inline deploy/klusterlet/olm-catalog/klusterlet/manifests/0000_00_operator.open-cluster-management.io_klusterlets.crd.yaml~Adopt new API updates for operator]: I think this file should be removed.

File: deploy/klusterlet/crds/0000_00_operator.open-cluster-management.io_klusterlets.crd.yaml~eeb95e939f074ff44e80dff94ef9973d34d8d28a [hunk 3]
```diff
@@ -0,0 +1,126 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (qiujian16) [inline deploy/klusterlet/crds/0000_00_operator.open-cluster-management.io_klusterlets.crd.yaml~eeb95e939f074ff44e80dff94ef9973d34d8d28a]: I think this file should be removed.

File: deploy/klusterlet/crds/0000_00_operator.open-cluster-management.io_klusterlets.crd.yaml~Adopt new API updates for operator [hunk 4]
```diff
@@ -0,0 +1,126 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (qiujian16) [inline deploy/klusterlet/crds/0000_00_operator.open-cluster-management.io_klusterlets.crd.yaml~Adopt new API updates for operator]: I think this file should be removed.

File: pkg/operators/klusterlet/bindata/bindata.go [hunk 5]
```diff
@@ -402,7 +402,7 @@ spec:
     spec:
       serviceAccountName: {{ .KlusterletName }}-work-sa
       containers:
-      - name: spoke-agent
+      - name: ç
```
Comment (qiujian16) [inline pkg/operators/klusterlet/bindata/bindata.go]: name here seems a typo?

File: manifests/klusterlet/klusterlet-work-deployment.yaml [hunk 6]
```diff
@@ -17,7 +17,7 @@ spec:
     spec:
       serviceAccountName: {{ .KlusterletName }}-work-sa
       containers:
-      - name: spoke-agent
+      - name: ç
```
Comment (qiujian16) [inline manifests/klusterlet/klusterlet-work-deployment.yaml]: name here seems a typo?

**Comment by mdelder** [comment]
/retest 


> 2020/06/04 20:45:57 Ran for 26m38s
> error: some steps failed:
>   * could not run steps: step e2e failed: failed to acquire lease: status 503 Service Unavailable, status code 503

**Pull request merged**